Packages/Signer/Sources/Chains/SuiSigner.swift

```diff
@@ -33,32 +37,27 @@ public struct SuiSigner: Signable {
     }
     
     func sign(input: SignerInput, privateKey: Data) throws -> String {
-        return signTxDataDigest(data: try input.metadata.getMessageBytes(), privateKey: privateKey)
+        return try signTxDataDigest(data: try input.metadata.getMessageBytes(), privateKey: privateKey)
     }
     
-    func signMessageBytes(coinType: CoinType, bytes: String, privateKey: Data) -> String {
-        let signingInput = SuiSigningInput.with {
-            $0.transactionPayload = .signDirectMessage(SuiSignDirect.with {
-                $0.unsignedTxMsg = bytes
-            })
-            $0.privateKey = privateKey
+    public func signMessage(message: SignMessage, privateKey: Data) throws -> String {
+        guard case .raw(let messageData) = message else {
+            throw AnyError("Sui message signing expects raw message bytes")
         }
-        
-        let output: SuiSigningOutput = AnySigner.sign(input: signingInput, coin: coinType)
-        return bytes + "_" + output.signature
+        let signer = GemstoneSigner()
+        return try signer.signSuiPersonalMessage(message: messageData, privateKey: privateKey)
     }
     
-    func signTxDataDigest(data: String, privateKey: Data) -> String {
-        let key = PrivateKey(data: privateKey)!
-        let pubKey = key.getPublicKeyEd25519()
-        
+    func signTxDataDigest(data: String, privateKey: Data) throws -> String {
         let parts = data.split(separator: "_").map { String($0) }
-        let digest = Data(hexString: parts[1])!
-        let signature = key.sign(digest: digest, curve: .ed25519)!
-        
-        var sig = Data([0x0])
-        sig.append(signature)
-        sig.append(pubKey.data)
-        return parts[0] + "_" + sig.base64EncodedString()
+        guard parts.count == 2 else {
+            throw AnyError("Invalid Sui digest payload")
+        }
+        guard let digest = Data(hexString: parts[1]) else {
+            throw AnyError("Invalid digest hex for Sui transaction")
+        }
+        let signer = GemstoneSigner()
+        let signature = try signer.signSuiDigest(digest: digest, privateKey: privateKey)
```

Comment:
```suggestion
        let signature = try GemstoneSigner().signSuiDigest(digest: digest, privateKey: privateKey)
```

since you only use once